src/utilities/finalisation.js

```diff
@@ -0,0 +1,25 @@
+/* eslint-disable import/prefer-default-export */
+/**
+ * mSupply Mobile
+ * Sustainable Solutions (NZ) Ltd. 2019
+ */
+
+import { modalStrings } from '../localization';
+
+/**
+ * Check whether a given customer invoice is safe to be finalised. If safe to finalise,
+ * return null, else return an appropriate error message.
+ *
+ * @param   {object}  customerInvoice  The customer invoice to check.
+ * @return  {string}                   Error message if unsafe to finalise, else null.
+ */
+export function checkForCustomerInvoiceError(customerInvoice) {
+  if (customerInvoice.items.length === 0) {
+    return modalStrings.add_at_least_one_item_before_finalising;
+  }
+
+  if (customerInvoice.totalQuantity === 0) {
+    return modalStrings.record_stock_to_issue_before_finalising;
+  }
+  return null;
+}
```

Comment:
Thought you were going to make this a class method on transaction? Though in terms of what is callable where I can understand keeping it out (or you'd have to check this isn't being called on transaction with type supplier invoice)

Reply:
yeah - I want to really badly :( I didn't want to put the method 'as is', at that would mean returning strings, so that would mean changing `FinaliseModal`.  As above I thought it would be more suitable to properly refactor/rewrite this when we need to fix the overall render issues (which when you look deeper into it will include changing up `NavigationBar`, `FinalizeModal` and `SyncModal`). Happy to put it back where it was if you think that's better!

Reply:
Nah I'm still happy with this!